tests/test_bootloader.py

```diff
@@ -0,0 +1,362 @@
+#!/usr/bin/python
+
+import unittest
+import sys
+import time
+import signal
+
+from intelhex import IntelHex
+
+from piksi_tools import serial_link
+from piksi_tools import flash
+from piksi_tools.bootload import Bootloader
+from piksi_tools.console.update_downloader import UpdateDownloader
+
+from sbp.client.handler import Handler
+
+from sbp.piksi  import SBP_MSG_RESET
+from sbp.system import SBP_MSG_HEARTBEAT
+
+# VCP to communicate with Piksi Under Test.
+PORT1 = None
+# VCP to communicate with second Piksi connected via UART to Piksi Under Test.
+PORT2 = None
+
+VERBOSE = False
+
+# Firmware to flash in tests.
+STM_FW_URL = \
+  "http://downloads.swiftnav.com/piksi_v2.3.1/stm_fw/piksi_firmware_v0.17.hex"
+NAP_FW_URL = \
+  "http://downloads.swiftnav.com/piksi_v2.3.1/nap_fw/swift_nap_v0.13.hex"
+STM_FW = None
+NAP_FW = None
+
+class Heartbeat():
+  """
+  Handle receiving heartbeat messages from Piksi. If we receive a heartbeat
+  from Piksi, we know that Piksi is in the application firmware.
+  """
+  def __init__(self, link):
+    self.received = False
+    self.link = link
```

Comment:
I think it should as the heartbeat message now sends the versioning info which needs to be parsed.